src/app/(profile)/_actions/countProfileView.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  getProfileById,
+  updateProfileById,
+} from '@/backend/profile/profile.service'
+import { withSentry } from '@/utils/errHandling'
+import { Prisma } from '@prisma/client'
+
+export const countProfileViews = withSentry(
+  async (profileId: string, payload: Prisma.ProfileUpdateInput) => {
+    const foundProfile = await getProfileById(profileId)
+
+    if (!foundProfile) {
+      throw new Error('Profile not found')
+    }
+
+    payload.viewCount = foundProfile.viewCount + 1
```

Comment:
**nit:** we should not mutate object this way but we should use prisma update query, where we have increment syntax: https://www.prisma.io/docs/orm/reference/prisma-client-reference#atomic-number-operations